### PR TITLE
[GraphBolt] Update multi-gpu example for regression benchmark.

### PR DESCRIPTION
**Pull request by RamonZhou**
## Description
Modified this example to get prepared for regression benchmark.
If a meticulous comparison between the graphbolt version and the DGL version is needed, then the DGL example might have to be updated as well.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/multigpu/graphbolt/node_classification.py [hunk 1]
```diff
@@ -393,6 +404,9 @@ def parse_args():
     # Load and preprocess dataset.
     dataset = gb.BuiltinDataset("ogbn-products").load()
 
+    # Thread limiting to avoid resource competition.
+    os.environ["OMP_NUM_THREADS"] = str(mp.cpu_count() // 2 // world_size)
```
Comment (peizhou001) [inline examples/multigpu/graphbolt/node_classification.py:408]: torch.set_num_threads(mp.cpu_count() // 2 // world_size)

**Review by peizhou001 — APPROVED** [review]

File: examples/multigpu/graphbolt/node_classification.py [hunk 2]
```diff
@@ -393,6 +404,10 @@ def parse_args():
     # Load and preprocess dataset.
     dataset = gb.BuiltinDataset("ogbn-products").load()
 
+    # Thread limiting to avoid resource competition.
```
Comment (frozenbugs) [inline examples/multigpu/graphbolt/node_classification.py:407]: @peizhou001 should this done in this PR or somewhere in graphbolt library?

**Review by frozenbugs — APPROVED** [review]

File: examples/multigpu/graphbolt/node_classification.py (hunk 2 above)
Comment (peizhou001) [reply examples/multigpu/graphbolt/node_classification.py:407]: Should be in this PR.

File: examples/multigpu/graphbolt/node_classification.py (hunk 1 above)
Comment (peizhou001) [reply examples/multigpu/graphbolt/node_classification.py:408]: chang back to 
` os.environ["OMP_NUM_THREADS"] = str(mp.cpu_count() // 2 // world_size)`
torch.set_num_threads seems doesn't work before spwan.

File: examples/multigpu/graphbolt/node_classification.py [hunk 3]
```diff
@@ -342,7 +350,7 @@ def run(rank, world_size, args, devices, dataset):
     )
     dist.reduce(tensor=test_acc, dst=0)
     if rank == 0:
-        print(f"Test Accuracy is {test_acc.item():.4f}")
+        print(f"Test Accuracy {test_acc.item():.4f}")
```
Comment (Rhett-Ying) [inline examples/multigpu/graphbolt/node_classification.py:353]: fix the `Test Accurayc: 4.55xxx` issue as well? or it's not an issue

**Pull request merged**